From b1b41fd250028051dc53764af366701cb844dc0d Mon Sep 17 00:00:00 2001
From: Ryan Zezeski <rpz@joyent.com>
Date: Sat, 24 Sep 2016 15:04:57 -0400
Subject: [PATCH] OS-5666 statechange lock_file() use of trap could be less
 confusing Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com> Approved by:
 Jerry Jelinek <jerry.jelinek@joyent.com>

---
 overlay/generic/usr/lib/brand/jcommon/poststate   | 2 +-
 overlay/generic/usr/lib/brand/jcommon/prestate    | 2 +-
 overlay/generic/usr/lib/brand/jcommon/statechange | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/overlay/generic/usr/lib/brand/jcommon/poststate b/overlay/generic/usr/lib/brand/jcommon/poststate
index cde9db9b..4e4b1207 100644
--- a/overlay/generic/usr/lib/brand/jcommon/poststate
+++ b/overlay/generic/usr/lib/brand/jcommon/poststate
@@ -25,7 +25,7 @@ if [[ -n $_ZONEADMD_brand_debug ]]; then
 	echo "zone $1 post-state-change $3 $4" >>$logfile
 	ksh -x /usr/lib/brand/$ps_brand/statechange "post" $@ 2>>$logfile
 	res=$?
-	echo "zone $1 post-state-change result $?" >>$logfile
+	echo "zone $1 post-state-change result $res" >>$logfile
 else
 	/usr/lib/brand/$ps_brand/statechange "post" $@
 	res=$?
diff --git a/overlay/generic/usr/lib/brand/jcommon/prestate b/overlay/generic/usr/lib/brand/jcommon/prestate
index 5fa95496..a61af897 100644
--- a/overlay/generic/usr/lib/brand/jcommon/prestate
+++ b/overlay/generic/usr/lib/brand/jcommon/prestate
@@ -25,7 +25,7 @@ if [[ -n $_ZONEADMD_brand_debug ]]; then
 	echo "zone $1 pre-state-change $3 $4" >>$logfile
 	ksh -x /usr/lib/brand/$ps_brand/statechange "pre" $@ 2>>$logfile
 	res=$?
-	echo "zone $1 pre-state-change result $?" >>$logfile
+	echo "zone $1 pre-state-change result $res" >>$logfile
 else
 	/usr/lib/brand/$ps_brand/statechange "pre" $@
 	res=$?
diff --git a/overlay/generic/usr/lib/brand/jcommon/statechange b/overlay/generic/usr/lib/brand/jcommon/statechange
index 0ad01040..339510fc 100644
--- a/overlay/generic/usr/lib/brand/jcommon/statechange
+++ b/overlay/generic/usr/lib/brand/jcommon/statechange
@@ -85,7 +85,7 @@ lock_file()
 	typeset prev_pid=0
 	while true; do
 		if (set -o noclobber; echo "$$" >$LOCKFILE) 2>/dev/null; then
-			trap 'ecode=$? rm -f $LOCKFILE; exit $ecode' INT TERM EXIT
+			trap 'rm -f $LOCKFILE' EXIT
 			break;
 		fi
 
-- 
2.21.0

